packages/apps/explorer/src/components/select-network/utils.ts

```diff
@@ -0,0 +1,42 @@
+import type { INetwork } from '@/context/networks-context';
+
+export const validateNewNetwork = (
+  networks: INetwork[],
+  newNetwork: INetwork,
+): string[] => {
+  const errors = [];
+
+  if (networks.some((network) => network.networkId === newNetwork.networkId)) {
+    errors.push('network already exists');
+  }
+
```

Comment:
I think I made this. Now that I think about it, we should probably check uniqueness on the combination of networkId and label. NetworkId can be used more than once, when you have multiple devnnets it'll be `development` for both.

Reply:
makes sense